.github/workflows/build-docs.yaml

```diff
@@ -0,0 +1,56 @@
+name: build-docs
+
+concurrency:
+  group: docs-build-on-${{ github.event_name }}-from-${{ github.ref_name }}
+  cancel-in-progress: true
+
+on:
+  # run only when called by other workflows
+  workflow_call:
+    inputs:
+      deploy:
+        required: true
+        type: boolean
+        default: false
+        description: "set to true to publish docs"
+
+jobs:
+  build:
+    runs-on: macos-latest
```

Comment:
Proposing using macOS because CRAN has pre-built binaries there. So CI is faster and less likely to fail with could-not-build-some-dependency-from-source types of errors.

The docs build takes around 3 minutes to run on macOS: https://github.com/uptake/uptasticsearch/actions/runs/13212614464/job/36888065982?pr=244